src/codeu/chat/client/Controller.java

```diff
@@ -46,13 +48,17 @@ public Message newMessage(Uuid author, Uuid conversation, String body) {
 
     try (final Connection connection = source.connect()) {
 
-      Serializers.INTEGER.write(connection.out(), NetworkCode.NEW_MESSAGE_REQUEST);
-      Uuid.SERIALIZER.write(connection.out(), author);
-      Uuid.SERIALIZER.write(connection.out(), conversation);
-      Serializers.STRING.write(connection.out(), body);
+      PrintWriter out = new PrintWriter(connection.out(), true);
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.in()));
+      
```

Comment:
Overall good change so far to use println and readln so you don't have to count bytes.  An extension of this work that makes sense to me is to move towards line-break as a message delimiter instead of a message-variable delimiter.  As in, find a way to pack an entire message (whether it's new_message.body+author+conversation or something else) into a single line.  That way, if you add a uuid of some sort fo yoru request/response pairs, the request -> response mechanism can be asynchronous.